include/SDL_log.h

```diff
@@ -53,6 +53,13 @@ extern "C" {
  */
 #define SDL_MAX_LOG_MESSAGE 4096
 
+/**
+ *  \brief The maximum size of temporary stack buffers
+ *
+ *  Messages shorter than this MAY be allocated on the stack.
+ */
+#define SDL_MAX_LOG_MESSAGE_STACK 512
+
```

Comment:
This shouldn't be in the public headers. Just put the define at the top of SDL_log.c.

Reply:
Also, let's make this like 128 (maybe?), under the assumption that most log messages are pretty short, and let malloc catch the ones that are uncommonly long. Less stack is better than more, since threads with extremely tiny stacks might want to log info.

Reply:
Using my `syslog` as a stand-in is perhaps not very representative, but it's what I have on hand.

~11.7% of log lines are >=128 characters and 2.2% are >=256

```
# awk '{++a[length() >= 128]} END {for (i in a) print i, a[i]}' syslog
0 2388
1 318
# awk '{++a[length() >= 256]} END {for (i in a) print i, a[i]}' syslog
0 2645
1 61
# awk '{++a[length() >= 512]} END {for (i in a) print i, a[i]}' syslog
0 2706
```

If you want 128, I'll switch to using the `SDL_MAX_SMALL_ALLOC_STACKSIZE` define.

Reply:
Let's do 256 for now.